tests/Unit/DataStructures/Test_DataVector.py

```diff
@@ -242,124 +219,119 @@ def test_hypot(self):
         self.assertEqual(a.max(), 5.0)
-        b = DataVector(2, 1.0)
-        b[1] = 0.1
+        b = DataVector([1.0, 0.5])
         self.assertEqual(b.max(), 1.0)
 
     def test_min(self):
-        a = DataVector(5, 4.0)
-        a[1] = 2.0
+        a = DataVector([4.0, 2.0, 4.0, 4.0, 4.0])
         self.assertEqual(a.min(), 2.0)
-        b = DataVector(2, 1.0)
-        b[1] = -0.5
+        b = DataVector([1.0, -0.5])
         self.assertEqual(b.min(),-0.5)
 
     def test_pow(self):
         a = DataVector(5, 4.0)
         self.assertEqual(a.pow(2), DataVector(5, 16.0))
-        b = DataVector(2, 1.0)
-        b[1] = -0.5
+        b = DataVector([1.0, -0.5])
         self.assertEqual(b.pow(2)[0], 1.0)
         self.assertEqual(b.pow(2)[1], 0.25)
 
     def test_sin(self):
         a = DataVector(5, 6.7)
         self.assertEqual(a.sin(),DataVector(5, math.sin(6.7)))
-        b = DataVector(2, 0.0)
-        b[1] = -0.5
+        b = DataVector([0.0, -0.5])
         self.assertEqual(b.sin()[0], 0.0)
         self.assertEqual(b.sin()[1], math.sin(-0.5))
 
     def test_sinh(self):
         a = DataVector(5, 3.0)
         self.assertEqual(a.sinh(), DataVector(5, math.sinh(3.0)))
-        b = DataVector(2, 0.0)
-        b[1] = -0.5
+        b = DataVector([0.0, -0.5])
         self.assertEqual(b.sinh()[0], 0.0)
         self.assertEqual(b.sinh()[1], math.sinh(-0.5))
 
     def test_sqrt(self):
         a = DataVector(5, 4.0)
         self.assertEqual(a.sqrt(), DataVector(5, 2.0))
-        b = DataVector(2, 1.0)
-        b[1] = 0.25
+        b = DataVector([1.0, 0.25])
         self.assertEqual(b.sqrt()[0], 1.0)
         self.assertEqual(b.sqrt()[1], 0.5)
 
     def test_step_function(self):
         a = DataVector(5, 4.0)
         self.assertEqual(a.step_function(),DataVector(5, 1.0))
-        b = DataVector(2, 1.0)
-        b[1] = -0.5
+        b = DataVector([1.0, -0.5])
         self.assertEqual(b.step_function()[0], 1.0)
         self.assertEqual(b.step_function()[1], 0.0)
 
     def test_tan(self):
         a = DataVector(5, 4.0)
         self.assertEqual(a.tan(), DataVector(5, math.tan(4.0)))
-        b = DataVector(2, 0.0)
-        b[1] = -0.5
+        b = DataVector([0.0, -0.5])
         self.assertEqual(b.tan()[0], 0.0)
         self.assertEqual(b.tan()[1], math.tan(-0.5))
 
     def test_tanh(self):
         a = DataVector(5, 4.0)
         self.assertEqual(a.tanh(), DataVector(5, math.tanh(4.0)))
-        b = DataVector(2, 0.0)
+        b = DataVector([0.0, -0.5])
         b[1] = -0.5
         self.assertEqual(b.tanh()[0], 0.0)
         self.assertEqual(b.tanh()[1], math.tanh(-0.5))
 
-    #Implicitly testing the iterator
+
     def test_numpy_compatibility(self):
         a = DataVector(5, 2.0)
         self.assertEqual(np.sin(a)[1], a.sin()[1])
+        b = np.array([1.0, 2.0, 3.0])
+        c = DataVector([1.0, 2.0, 3.0])
+        self.assertEqual((b + c).all(), np.array([2.0, 4.0, 6.0]).all())
```

Comment:
What is the `.all()` for? Should probably be removed

Reply:
Python says that equality of two arrays was ambiguous, and it must be explicitly stated whether if it is equality for any element or equality for all elements.  The test fails if it is removed.

Reply:
Either use `assertSequenceEqual` (preferred imo) or assert for `(b + c == np.array(...)).all()`. What you currently do reduces `(b + c)` to a boolean, which is true only if all elements are 1. So the test succeeds, but it doesn’t test what we want.